src/renderer/components/chat/ChatList.tsx

```diff
@@ -371,59 +382,129 @@ function useLogic(queryStr: string, showArchivedChats: boolean) {
   }>({})
 
   const isChatLoaded: (params: Index) => boolean = ({ index }) =>
-    !!chatLoadState[chatListIds[index]]
+    !!chatLoadState[chatListIds[index][0]]
   const loadChats: (params: IndexRange) => Promise<void> = async ({
     startIndex,
     stopIndex,
   }) => {
-    const chatIds = chatListIds.slice(startIndex, stopIndex + 1)
+    const entries = chatListIds.slice(startIndex, stopIndex + 1)
     setChatLoading(state => {
-      chatIds.forEach(id => (state[id] = LoadStatus.FETCHING))
+      entries.forEach(
+        ([chatId, _msgId]) => (state[chatId] = LoadStatus.FETCHING)
+      )
       return state
     })
     const chats = await DeltaBackend.call(
-      'chatList.getChatListItemsByIds',
-      chatIds
+      'chatList.getChatListItemsByEntries',
+      entries
     )
     setChatCache(cache => ({ ...cache, ...chats }))
     setChatLoading(state => {
-      chatIds.forEach(id => (state[id] = LoadStatus.LOADED))
+      entries.forEach(([chatId, _msgId]) => (state[chatId] = LoadStatus.LOADED))
       return state
     })
   }
 
-  const onChatListItemChanged = async (
+  const onChatListItemChanged = (
     _event: any,
-    { chatId }: { chatId: number }
+    [chatId, messageId]: [number, number]
   ) => {
-    if (chatId === 0) {
-      // setChatLoading({})
-      // setChatCache({})
-    } else {
+    const updateChatListItem = async (chatId: number, messageId: number) => {
       setChatLoading(state => ({
         ...state,
         [chatId]: LoadStatus.FETCHING,
       }))
-      const chats = await DeltaBackend.call('chatList.getChatListItemsByIds', [
-        chatId,
-      ])
+      const chats = await DeltaBackend.call(
+        'chatList.getChatListItemsByEntries',
+        [[chatId, messageId]]
+      )
       setChatCache(cache => ({ ...cache, ...chats }))
       setChatLoading(state => ({
         ...state,
         [chatId]: LoadStatus.LOADED,
       }))
     }
+    if (chatId === 0) {
+      // setChatLoading({})
+      // setChatCache({})
+    } else {
+      if (messageId === 0) {
+        // if no message id is provided it tries to take the old one
+        // workaround to get msgId
```

Comment:
I think I need to redo this workaround as it uses chatListIds which can and probably will be a deprecated snapshot of the value